src/index.js

```diff
@@ -58,22 +58,22 @@ const Alpine = {
         const observer = new MutationObserver((mutations) => {
             if (this.pauseMutationObserver) return;
 
-            for (let i=0; i < mutations.length; i++){
-                if (mutations[i].addedNodes.length > 0) {
-                    mutations[i].addedNodes.forEach(node => {
-                        // Discard non-element nodes (like line-breaks)
-                        if (node.nodeType !== 1) return
-
-                        // Discard any changes happening within an existing component.
-                        // They will take care of themselves.
-                        if (node.parentElement && node.parentElement.closest('[x-data]')) return
-
-                        this.discoverUninitializedComponents((el) => {
-                            this.initializeComponent(el)
-                        }, node.parentElement)
-                    })
-                }
-              }
+            mutations.forEach(mutation => {
```

Comment:
forEach is waaaay faster than our old-lovely and imperative for-loop.

Proof ([benchmark](https://www.measurethat.net/Benchmarks/Show/3862/array-loops-for-vs-for-in-vs-for-of-vs-foreach-vs-map-f))
![image](https://user-images.githubusercontent.com/4066435/85788722-1401ca00-b736-11ea-9fc7-940b46cede37.png)